### PR TITLE
feature: Compose Multiplatform 1.6.10-beta01 Lifecycle and Navigation libraries

### PR DESCRIPTION
**Pull request by zamulla**
Added pages detailing experimental libraries for Lifecycle and Navigation in the 1.6.10-beta01 release.

### CODE REVIEW

File: topics/compose/compose-lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+Lifecycle of components in Compose Multiplatform is adopted from the Jetpack Compose [Lifecycle](https://developer.android.com/topic/libraries/architecture/lifecycle)
+concept. Lifecycle-aware components can react to changes in the lifecycle status of other components and help you
+produce better-organized, and often lighter code, that is easier to maintain.
+
+## Multiplatform lifecycle
+
+Lifecycle functionality on Android originally was represented by [a set of callbacks](https://developer.android.com/guide/components/activities/activity-lifecycle).
+Android Jetpack introduced [the dedicated library](https://developer.android.com/reference/kotlin/androidx/lifecycle/package-summary.html)
+with the `Lifecycle` class to observe activity states.
```
Comment (MatkovIvan) [inline topics/compose/compose-lifecycle.md:11]: Android only statement in "Multiplatform" section

File: topics/compose/compose-lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+![Lifecycle diagram](lifecycle-states.svg){width="700"}
+
+### Mapping Android lifecycle to other platforms
+
+#### iOS
+
+| Native event/notification | Lifecycle event | Lifecycle state change |
+|---------------------------|-----------------|------------------------|
+| viewDidDisappear          | ON_STOP         | STARTED → CREATED      |
```
Comment (MatkovIvan) [inline topics/compose/compose-lifecycle.md:25]: Put all code names in `
```suggestion
| `viewDidDisappear`          | `ON_STOP`         | `STARTED` → `CREATED`      |
```

File: topics/compose/compose-navigation-routing.md [hunk 3]
```diff
@@ -1,15 +1,77 @@
...
+    1. Choose the starting destination from the list of routes you defined earlier.
+    2. Create a navigation graph, either directly, as part of creating a `NavHost`, or programmatically, using the
+       `NavController.createGraph()` function.
+
+Each back stack entry (each navigation route included in the graph) implements the `LifecycleOwner` interface.
+A switch between different screens of the app makes it change its state from `RESUMED` to `STARTED` and back.
+`RESUMED` is also described as "settled": the act of navigation is considered finished when the new screen is prepared and active.
+See the [Lifecycle](compose-lifecycle.md) page for details on the current lifecycle implementation in Compose Multiplatform.
+
+Things to note:
```
Comment (danil-pavlov) [inline topics/compose/compose-navigation-routing.md:52]: Feels a little abrupt, some alternatives for the intro phrase: "Also, keep in mind that: ", "It's also important to note that:", "Also, consider the following implementation aspects:"

File: topics/compose/compose-lifecycle.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+| `viewDidDisappear`                   | `ON_STOP`       | `STARTED` → `CREATED`  |
+| `viewWillAppear`                     | `ON_START`      | `CREATED` → `STARTED`  |
+| `didEnterBackground`                 | `ON_PAUSE`      | `RESUMED` → `STARTED`  |
+| `willEnterForeground`                | `ON_RESUME`     | `STARTED` → `RESUMED`  |
+
+### Web
+
+Due to limitations of the Wasm target, lifecycles:
+
+* skip the `CREATED` state as the application is always attached to the page,
```
Comment (danil-pavlov) [inline topics/compose/compose-lifecycle.md:38]:
```suggestion
* Skip the `CREATED` state, as the application is always attached to the page.
```

**Review by danil-pavlov — CHANGES_REQUESTED** [review]

File: topics/compose/compose-lifecycle.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
+[//]: # (title: Lifecycle)
+
+Lifecycle of components in Compose Multiplatform is adopted from the Jetpack Compose [lifecycle](https://developer.android.com/topic/libraries/architecture/lifecycle)
+concept.
+Lifecycle-aware components can react to changes in the lifecycle state of other components and help you
+produce better-organized, and often lighter, code that is easier to maintain.
+
+Lifecycle functionality on Android originally consisted of [a set of callbacks](https://developer.android.com/guide/components/activities/activity-lifecycle).
```
[the remark this answers is not in the document]
Comment (zamulla) [reply topics/compose/compose-lifecycle.md]: I just like history. Makes sense, squished all of that down.

**Pull request merged**